R-packages/modeltools/R/matrix.R

```diff
@@ -0,0 +1,75 @@
+#' Create a training and prediction matrices for a given ahead.
+#'
+#' @param lagged_df Data frame of lagged data.  It should have the following columns:
+#'     \itemize{
+#'     \item{`geo_value`}{Strings of geographic locations.}
+#'     \item{`time_value`}{Dates of training data.}
+#'     \item{Covariate columns}{Columns with names of the form `value-{days}:{signal}` whose values
+#'         correspond to `{signal}` `{days}` before `time_value`}
+#'     \item{Response columns}{Columns with names of the form `value+{n}:{response}` whose values
+#'         correspond to `{response}` `{n}` incidence period units after `time_value`.  Since
+#'         columns beginning with "value+" are interpretted as the response rather than a
+#'         covariate, observed data from the same day as `time_value` should be in columns
+#'         `value-0:{signal}` rather than `value+0:{signal}`}
+#'     }
+#'     A data frame in this format can be made using `covidcast::aggregate_signals()`.
+#' @param ahead Number of incidence period units (i.e., epiweeks, days, etc.) ahead to forecast
+#' @param training_window_size Size of the local training window in days to use. For example, if
```

Comment:
The training window size is always in days because the signal variables are reported daily or aggregated over some period but associate with a single day.  This is because we have no way to express time values that are not days (i.e., "epiweek 33").  So `incidence_period` only applies to our interpretation of `ahead`s (future data), not to the downloaded signals (past data).

Reworded slightly to clarify.